binder/src/test/java/io/grpc/binder/RobolectricBinderSecurityTest.java

```diff
@@ -0,0 +1,482 @@
+      ComponentName componentName = new ComponentName(context, SomeService.class);
+      shadowOf(context)
+          .setComponentNameAndServiceForBindService(
+              componentName, checkNotNull(binderReceiver.get()));
+    }
+
+    /**
+     * Returns an {@link ScheduledExecutorService} under which all of the gRPC computations run. The
+     * execution of any pending tasks on this executor can be triggered via {@link #idleLooper()}.
+     */
+    ScheduledExecutorService getExecutor() {
+      return scheduledExecutorService;
+    }
+
+    void idleLooper() {
+      shadowOf(handlerThread.getLooper()).idle();
+    }
+
+    void setSecurityPolicyStatusWhenReady(Status status) {
+      Uninterruptibles.takeUninterruptibly(statusesToSet).set(status);
+    }
+
+    @Override
+    public IBinder onBind(Intent intent) {
+      super.onBind(intent);
+      return checkNotNull(binderReceiver.get());
+    }
+
+    @Override
+    public void onDestroy() {
+      super.onDestroy();
+      server.shutdownNow();
+      handlerThread.quit();
+    }
+
+    /** A future representing a task submitted to a {@link Handler}. */
+    private static class HandlerFuture<V> implements ScheduledFuture<V> {
+
+      private final Duration delay;
+      private final SettableFuture<V> delegate = SettableFuture.create();
+
+      HandlerFuture(Duration delay) {
+        this.delay = delay;
+      }
+
+      @Override
+      public long getDelay(TimeUnit timeUnit) {
+        return timeUnit.convert(delay.toMillis(), TimeUnit.MILLISECONDS);
+      }
+
+      @Override
+      public int compareTo(Delayed other) {
+        return Comparator
+            .comparingLong((Delayed delayed) -> delayed.getDelay(TimeUnit.MILLISECONDS))
+            .compare(this, other);
+      }
+
+      @Override
+      public boolean cancel(boolean mayInterruptIfRunning) {
+        return delegate.cancel(mayInterruptIfRunning);
+      }
+
+      @Override
+      public boolean isCancelled() {
+        return delegate.isCancelled();
+      }
+
+      @Override
+      public boolean isDone() {
+        return delegate.isDone();
+      }
+
+      @Override
+      public V get() throws ExecutionException, InterruptedException {
+        return delegate.get();
+      }
+
+      @Override
+      public V get(long timeout, TimeUnit timeUnit)
+          throws ExecutionException, InterruptedException, TimeoutException {
+        return delegate.get(timeout, timeUnit);
+      }
+
+      void complete(V result) {
+        delegate.set(result);
+      }
+
+      void setException(Exception e) {
+        delegate.setException(e);
+      }
+    }
+
+    /**
+     * Minimal implementation of a {@link ScheduledExecutorService} that delegates tasks to a
+     * {@link Handler}. Pending tasks can be forced to run via {@link #idleLooper()}.
+     */
+    private class HandlerScheduledExecutorService implements ScheduledExecutorService {
```

Comment:
Could you extend AbstractExecutorService to avoid re-implementing some of the methods common to any ExecutorService?

Reply:
Done. Got rid of the `invoke` and `submit` methods.